### PR TITLE
avm2: Add X509 certificate classes

### PR DESCRIPTION
**Pull request by onkrot**

### CODE REVIEW

**Comment by Herschel** [comment]
You can use `__ruffle__.stub_getter` to indicate that these are unimplemented ([example](https://github.com/ruffle-rs/ruffle/blob/47f594d1080e2fe54cc6f853dc5f64adca4475b4/core/src/avm2/globals/flash/media/Microphone.as#L6))

File: core/src/avm2/globals/flash/security/X500DistinguishedName.as [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package flash.security
+{
+    public final class X500DistinguishedName
```
Comment (Herschel) [inline core/src/avm2/globals/flash/security/X500DistinguishedName.as:3]: Looks like this class has a `toString` method as well

**Comment by Lord-McSweeney** [comment]
I don't think `stub_getter` is needed for these since the current implementation is technically correct- it just needs to be hooked up to `SecureSocket` if that (ever) is implemented.

**Comment by Herschel** [comment]
> I don't think `stub_getter` is needed for these since the current implementation is technically correct- it just needs to be hooked up to `SecureSocket` if that (ever) is implemented.

I double checked, and I agree -- I was worried that these getters should return some default value or throw on a default instance, but just checked in Flash and looks like these indeed return `null`s if you construct your own instance.

(Funnily doing `new X509Certificate().encoded` crashes Flash CS6 immediately!)

**Pull request closed without merging**